webapp/src/route.tsx

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react'
+import {
+    Redirect,
+    Route,
+    useRouteMatch,
+} from 'react-router-dom'
+
+import {Utils} from './utils'
+import {getLoggedIn} from './store/users'
+import {useAppSelector} from './store/hooks'
+import {UserSettings} from './userSettings'
+
+type RouteProps = {
+    path: string|string[]
+    exact?: boolean
+    render?: (props: any) => React.ReactElement
+    children?: React.ReactElement
+    getOriginalPath?: (match: any) => string
+    loginRequired?: boolean
+}
+
+function FBRoute(props: RouteProps) {
```

Comment:
I create a new concept of route, the FBRoute, this is just a route that accept some special parameters: `loginRequired` that forces to redirect to login when the user is not logged in, and the `getOriginalPath` that give you the redirect url for login and welcome pages.

This component ensure that if you are logged in but you haven't seen yet the welcome page, it is going to redirect you there, and with a redirect page if is needed.

Same for the login machinery.